packages/design-system/src/scalars/components/testing-form/form.tsx

```diff
@@ -0,0 +1,103 @@
+import { ReactNode } from "react";
+import {
+  FieldValues,
+  UseFormReturn,
+  useForm,
+  SubmitHandler,
+} from "react-hook-form";
+import { StringField } from "../string-field";
+import { BooleanField } from "../boolean-field";
+
+interface FormProps<T extends FieldValues> {
+  children: (methods: UseFormReturn<T>) => ReactNode;
+  onSubmit: SubmitHandler<T>;
+  defaultValues?: Partial<T>;
+  mode?: "onBlur" | "onChange" | "onSubmit" | "onTouched" | "all";
+}
+
+export const Form: React.FC<FormProps<FieldValues>> = ({
+  children,
+  onSubmit,
+  defaultValues,
+  mode = "onSubmit",
+}) => {
+  const methods = useForm({
+    defaultValues,
+    mode,
+  });
+
+  return (
+    <form onSubmit={methods.handleSubmit(onSubmit)}>{children(methods)}</form>
+  );
+};
+
+export const FormExample = () => {
+  type FormData = {
+    email: string;
+    name: string;
+    agree: boolean;
+  };
+
+  const onSubmit = (data: FormData) => {
+    console.log("Form submitted:", data);
+  };
+
+  return (
+    <Form onSubmit={onSubmit}>
+      {({ register, formState: { errors } }) => (
+        <div className="w-96 space-y-4 rounded-lg bg-white p-6 shadow-md">
+          <h2 className="text-xl font-semibold">Contact Form</h2>
+
+          <StringField
+            label="Name"
+            placeholder="Enter your name"
+            required
+            {...register("name", { required: "Name is required" })}
```

Comment:
I am in doubt, the component StringField, is already built using  Controller, and wrapper, so why does it register input again
https://react-hook-form.com/docs/usecontroller